mkosi/__init__.py

```diff
@@ -3961,6 +3961,16 @@ def insert_partition(
             run(["blkdiscard", path])
             path.write_bytes(blob.read())
 
+            # Required otherwise the partition removal will fail
+            if dev is not None:
+                with open(loopdev, 'rb+') as f:
+                    ioctl_partition_add(
+                        f.fileno(),
+                        part.number,
+                        args.partition_table.partition_offset(part),
+                        args.partition_table.partition_size(part)
+                    )
```

Comment:
This should also be moved up so that it's executed right after we run `run_sfdisk()`